### ISSUE TITLE
file access permission issue with rmlint 2.10.1

### ISSUE THREAD

**Issue by chenrui333** [sahib/rmlint#438]
It looks like the file permission check behavior got changed and broke the checks in Mojava (osx 10.14) and high-sierra (osx 10.13) builds.

```
$ /usr/local/Cellar/rmlint/2.10.1/bin/rmlint
WARNING: Can't open directory or file "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/": Invalid argument
==> 0 file(s) after investigation, nothing to search through.
```

relates to https://github.com/Homebrew/homebrew-core/pull/58249 https://github.com/Homebrew/homebrew-core/pull/61404

**Comment by Sarkasper** [comment]
I have the same issue in void linux:

>  rmlint -g .
WARNING: Can't open directory or file ".": Invalid argument
ERROR: Not all given paths are valid. Aborting.

**Comment by Sarkasper** [comment]
I have the same issue in Void Linux (x86_64-musl):

>  rmlint -g .
WARNING: Can't open directory or file ".": Invalid argument
ERROR: Not all given paths are valid. Aborting.

**Comment by zokker13** [comment]
I second this on freebsd as well

**Comment by ChrisBaker97** [comment]
For reference, this breaking change seems to have happened between `2.9.0` and `2.10.0`. (Installed with Homebrew on macOS 10.14.6 Mojave.)
```
$ echo "1" > 1.txt > 2.txt
```
On version `2.0.9`:
```
$ rmlint --version && rmlint
version 2.9.0 compiled: Nov 12 2020 at [13:03:33] "Odd Olm" (rev unknown)
compiled with: -mounts -nonstripped -fiemap +sha512 +bigfiles +intl +replay +xattr -btrfs-support

rmlint was written by Christopher <sahib> Pahl and Daniel <SeeSpotRun> Thomas.
The code at https://github.com/sahib/rmlint is licensed under the terms of the GPLv3.

# Duplicate(s):
    ls '/Users/chris/test/1.txt'
    rm '/Users/chris/test/2.txt'

==> Note: Please use the saved script below for removal, not the above output.
==> In total 3 files, whereof 1 are duplicates in 1 groups.
==> This equals 2 B of duplicates which could be removed.
==> Scanning took in total 0.137s.

Wrote a sh file to: /Users/chris/test/rmlint.sh
Wrote a json file to: /Users/chris/test/rmlint.json
```
On version `2.10.0`:
```
$ rmlint --version && rmlint
version 2.10.0 compiled: Nov 12 2020 at [13:10:45] "Ludicrous Lemur" (rev unknown)
compiled with: -mounts -nonstripped -fiemap +sha512 +bigfiles +intl +replay +xattr -btrfs-support

rmlint was written by Christopher <sahib> Pahl and Daniel <SeeSpotRun> Thomas.
The code at https://github.com/sahib/rmlint is licensed under the terms of the GPLv3.
WARNING: Can't open directory or file "/Users/chris/test/": Invalid argument
==> 0 file(s) after investigation, nothing to search through.
```

**Comment by sahib** [comment]
Should be fixed as of a726d0c - change will be in the next release which will come soon-ish.
